src/longbow/extract/command.py

```diff
@@ -166,11 +129,53 @@ def main(pbi, out_file, force, base_padding, leading_adapter, trailing_adapter,
             total=read_count
             ) as pbar:
 
+        # Get our model:
+        if model is None:
+            model = bam_utils.get_model_name_from_bam_header(bam_file.header)
```

Comment:
It's possible that the user has specified a model from a JSON file.  In this case, the JSON of the model itself will be written to the bam header.

Can you add a check for this case and have it create the model from that JSON?

Reply:
Done - for every command except `annotate`, we now load the model directly from the BAM header by default.